packages/wrangler/src/index.tsx

```diff
@@ -71,6 +78,19 @@ async function readConfig(configPath?: string): Promise<Config> {
   // todo: validate, add defaults
   // let's just do some basics for now
 
+  for (const binding of config.unsafe?.bindings ?? []) {
+    if ("type" in binding && typeof binding["type"] == "string") {
+      const type = binding["type"];
+      if (knownBindings.includes(type)) {
+        console.warn(
+          `Raw '${type}' bindings are not directly supported by wrangler. Consider migrating to a ` +
+            `format for '${type}' bindings that is supported by wrangler for optimal support: ` +
+            "https://developers.cloudflare.com/workers/cli-wrangler/configuration"
+        );
+      }
+    }
+  }
+
```

Comment:
We should also log a warning if unsafe bindings are used at all, saying it may break at any time.